conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureAsyncRequestProcessing.java

```diff
@@ -133,18 +134,41 @@
         //    after this Endpoints HttpHandler returns. Otherwise an unlucky future could race
         //    HttpServerExchange.isInCall transitioning from true -> false, causing hte dispatch task not
         //    to execute.
+        DeferredTracer tracer = new DeferredTracer("Undertow: Async Result");
         exchange.dispatch(() -> Futures.addCallback(future, new FutureCallback<T>() {
             @Override
             public void onSuccess(@Nullable T result) {
-                exchange.dispatch(new ConjureExceptionHandler(serverExchange ->
-                        returnValueWriter.write(result, exchange), exceptionSerializer));
+                exchange.dispatch(wrapCallback(serverExchange -> returnValueWriter.write(result, exchange), tracer));
             }
 
             @Override
             public void onFailure(Throwable throwable) {
-                exchange.dispatch(new ConjureExceptionHandler(serverExchange -> ConjureExceptions.handle(
-                        serverExchange, exceptionSerializer, throwable), exceptionSerializer));
+                exchange.dispatch(wrapCallback(serverExchange -> ConjureExceptions.handle(
+                        serverExchange, exceptionSerializer, throwable), tracer));
             }
         }, DIRECT_EXECUTOR));
     }
+
+    private HttpHandler wrapCallback(HttpHandler action, DeferredTracer tracer) {
+        return new DeferredTracerHandler(new ConjureExceptionHandler(action, exceptionSerializer), tracer);
```

Comment:
Could we avoid the new class and return a lambda instead?
```suggestion
        return exchange -> tracer.withTrace(() -> {
            new ConjureExceptionHandler(action, exceptionSerializer).handleRequest(exchange);
            return null;
        });
```

Reply:
Updated. I find classes easier to follow in stack traces, but I don't have much preference either way.